schema/auth.json

```diff
@@ -0,0 +1,176 @@
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique auth name",
+          "minLength": 1
+        },
+        "scheme": {
+          "type": "string",
+          "description": "Defines the auth type",
+          "enum": [
+            "basic",
+            "bearer",
+            "oidc"
+          ],
+          "default": "basic"
+        },
+        "info": {
+          "oneOf": [
+            {
+              "type": "string",
+              "description": "Expression referencing a workflow secret that contains all needed auth info"
+            },
+            {
+              "title": "Basic Auth Info",
+              "$ref": "#/definitions/basicinfodef"
+            },
+            {
+              "title": "Bearer Auth Info State",
+              "$ref": "#/definitions/bearerinfodef"
+            },
+            {
+              "title": "Oidc Auth Info",
+              "$ref": "#/definitions/oidcinfodef"
+            }
+          ]
+        }
+      },
+      "required": [
+        "name",
+        "info",
+        "scheme"
+      ]
+    },
+    "basicinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed basic auth info"
+        },
+        {
+          "type": "object",
+          "description": "Basic auth information",
+          "properties": {
+            "username": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user name",
+              "minLength": 1
+            },
+            "password": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user password",
+              "minLength": 1
+            }
+          },
+          "required": [
+            "username",
+            "password"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "bearerinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed bearer auth info"
+        },
+        {
+          "type": "object",
+          "description": "Bearer auth information",
+          "properties": {
+            "token": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the token",
+              "minLength": 1
+            }
+          },
+          "required": [
+            "token"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "oidcinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed oidc auth info"
+        },
+        {
+          "type": "object",
+          "description": "Oidc auth information",
+          "properties": {
```

Comment:
I'm sorry I just realized while implementing it that I forgot some properties (mainly used by the [token-exchange](https://datatracker.ietf.org/doc/html/rfc8693) grant type):
- **audiences**: array of string
- **subjectToken**: string
- **requestedSubject**: string
- **requestedIssuer**: string

Reply:
added